dev/com.ibm.ws.kernel.service/src/com/ibm/ws/kernel/service/util/SecureAction.java

```diff
@@ -583,6 +584,20 @@ public ClassLoader getContextClassLoader() {
             return AccessController.doPrivileged(getContextClassLoaderAction, controlContext);
     }
 
+    public void setContextClassLoader(final ClassLoader classLoader) {
+        if (System.getSecurityManager() == null) {
+            Thread.currentThread().setContextClassLoader(classLoader);
+        } else {
+            AccessController.doPrivileged(new PrivilegedAction<Void>() {
+                @Override
+                public Void run() {
+                    Thread.currentThread().setContextClassLoader(classLoader);
+                    return null;
+                }
+            }, controlContext);
+        }
+    }
+
```

Comment:
Instead of adding this method I would prefer we use the existing API `com.ibm.ejs.util.dopriv.SetContextClassLoaderPrivileged` which is a much more performant implementation of setting the TCCL. 

It would be nice to have this function consolidated into the `SecureAction` class, so perhaps you could keep this method and delegate off to the existing API

Reply:
Ok, I’ll take a look at that tomorrow... and we should probably advertise things like this as best practice

Reply:
I have switched to using `com.ibm.ejs.util.dopriv.SetContextClassLoaderPrivileged`. I would have liked to have made the code in `com.ibm.ws.kernel.service` common too but that will have to be another issue at another time.